integration_tests/test_ibc.py

```diff
@@ -0,0 +1,159 @@
+import json
+import os
+import time
+from pathlib import Path
+
+import pytest
+
+from .network import setup_chainmain, setup_cronos, setup_hermes
+
+
+@pytest.fixture(scope="module")
+def cronos(request, tmp_path_factory):
+    "start-cronos"
+    yield from setup_cronos(tmp_path_factory.mktemp("cronos"), 26700, True)
+
+
+@pytest.fixture(scope="module")
+def chainmain(tmp_path_factory):
+    # "start-cronos"
+    yield from setup_chainmain(tmp_path_factory.mktemp("chainmain"), 26800)
+
+
+@pytest.fixture(scope="module")
+def hermes(tmp_path_factory):
+    time.sleep(20)
+    yield from setup_hermes(tmp_path_factory.mktemp("hermes"), 26900)
+
+
+def get_balance(chain, addr, denom):
+    output = chain.cosmos_cli(0).raw(
+        "query",
+        "bank",
+        "balances",
+        addr,
+        node=chain.node_rpc(0),
+        output="json",
+    )
+    c = json.loads(output.decode())
+    coins = c["balances"]
+    for coin in coins:
+        if coin["denom"] == denom:
+            value = int(coin["amount"])
+            return value
+    return 0
+
+
+def test_ibc(cronos, chainmain, hermes):
+    # wait for hermes
+    time.sleep(20)
+    while True:
+        my_ibc0 = "chainmain-1"
+        my_ibc1 = "cronos_777-1"
+        my_channel = "channel-0"
+        my_config = hermes.configpath
+        # signer21
+        coin_receiver = "crc1q04jewhxw4xxu3vlg3rc85240h9q7ns6hglz0g"
+        src_amount = 5
+        src_denom = "basecro"
+        dst_denom = "basetcro"
+        # dstchainid srcchainid srcportid srchannelid
+        # chainmain-1 -> cronos_777-1
+        cmd = f"hermes -c {my_config} tx raw ft-transfer \
+        {my_ibc1} {my_ibc0} transfer {my_channel} {src_amount} \
+        -o 1000 -n 1 -d {src_denom} -r {coin_receiver} -k testkey"
+        os.popen(cmd)
+        dstaddr = f"{coin_receiver}"
+        olddstbalance = get_balance(cronos, dstaddr, dst_denom)
+        time.sleep(5)
+        newdstbalance = get_balance(cronos, dstaddr, dst_denom)
+        expectedbalance = olddstbalance + src_amount * (10 ** 10)
+        assert expectedbalance == newdstbalance
+        break
+    assert True
+    pass
+
+
+def test_ibc_reverse(cronos, chainmain, hermes):
+    # wait for hermes
+    time.sleep(20)
+    while True:
+        my_ibc0 = "chainmain-1"
+        my_ibc1 = "cronos_777-1"
+        my_channel = "channel-0"
+        my_config = hermes.configpath
+        # signer21
+        coin_receiver = "cro1u08u5dvtnpmlpdq333uj9tcj75yceggszxpnsy"
+        src_amount = 2 * (10 ** 10)
+        src_denom = "basetcro"
+        dst_denom = "ibc/6B5A664BF0AF4F71B2F0BAA33141E2F1321242FBD\
+5D19762F541EC971ACB0865"
+        # dstchainid srcchainid srcportid srchannelid
+        # chainmain-1 <- cronos_777-1
+        cmd = f"hermes -c {my_config} tx raw ft-transfer \
```

Comment:
I think the better test would be to test the "transfer-tokens` command in cronos

src_denom = "basetcro"
dst_denom = "basecro"

check that olddstbalance + src_amount / (10 ** 10)

Reply:
i'l add as another pr